src/api/Browser.ts

```diff
@@ -0,0 +1,52 @@
+import EventEmitter from 'events'
+import { TSend } from '../protocol'
+import Page from './Page'
+
+class Browser extends EventEmitter {
+  private _send: TSend
+
+  constructor (arg: { send: TSend }) {
```

Comment:
instead of assigning the ctor argument locally you could write something like this:
```ts
constructor(private _send: TSent) { }
```
and you won't need the `private _send: TSend` field

Reply:
Oh… I'm afraid that this will not work because Foxr is using Babel 7 with TS preset to build, `tsc` is only for runtime checking. That behavior is totally not ES6+ compatible and comes from a legacy TS classes era – main reason I didn't want to touch classes at all.

ESLint with TS parser complains as well.

Thanks for the reference.

Reply:
any specific reason why you prefer using Babel 7 & typescript instead of pure typescript? both of them compile to clean js?

Reply:
I just got used to Babel ecosystem a lot for many years and believe that TypeScript should deal only with the runtime. Everything I make is using Babel.

Reply:
btw babel also knows how to transpile constructors:
```ts
export class Browser extends EventEmitter {

  constructor(private send: TSend) {
    super()
  }

  async close() {
    await this.send('Marionette:AcceptConnections', { value: false })
    await this.send('Marionette:Quit')

    this.emit('disconnected')
  }
...
```
babel output:
```js
export class Browser extends EventEmitter {
  constructor(send) {
    super();
    this.send = send;
  }

  close() {
    var _this = this;

    return _asyncToGenerator(function* () {
      yield _this.send('Marionette:AcceptConnections', {
        value: false
      });
      yield _this.send('Marionette:Quit');

      _this.emit('disconnected');
    })();
  }
...
```
notice the assignment in the ctor :)

Reply:
Looks a bit weird because in fact args are unused... I think I'll use the current way just because of personal opinion even if it's a bit more verbose.